sync/src/chain.rs

```diff
@@ -452,7 +453,11 @@ impl ChainSync {
 
 		// Disable the peer for this syncing round if it gives invalid chain
 		if !valid_response {
-			trace!(target: "sync", "{} Disabled for invalid headers response", peer_id);
+			trace!(target: "sync", "{} Deactivated for invalid headers response", peer_id);
+			self.deactivate_peer(io, peer_id); 
+		}
+		if headers.is_empty() {
```

Comment:
this is to handle the case when we're on a fork and try to seek the common block. I've added an additional check for the state for now. Will add validation for receiving blocks we did not ask for in the following PR.